libcob/common.c

```diff
@@ -2599,23 +2604,16 @@ cob_set_exception (const int id)
 		static char excp_para[COB_MAX_WORDLEN + 1];
 		cob_module	*mod = COB_MODULE_PTR;
 		cobglobptr->cob_got_exception = 1;
-#if 0	/* consider addition for 4.x */
 		cobglobptr->last_exception_source = cob_source_file;	/* needs to be strdup'd */
```

Comment:
this todo comment needs to be checked, especially as we free the memory down there in this PR

Reply:
Checked how ?

Reply:
Check its use. If `cobglobptr->last_exception_source` is used after the pointer that cob_source_file points to is freed (either directly or when it points to module static data the module may be unloaded via CANCEL), then we need an strdup+free; otherwise we don't need either of those.

Reply:
Well `last_exception_source` does not seem to be used anywhere for now, but since there is some code that freese it, I guess it has to be allocated, hence strdup'd.

BTW, the code that frees it is suspicious:
```
		if (cobglobptr->last_exception_source) {
			cobglobptr->last_exception_source = NULL;
			cob_free ((void *)cobglobptr->last_exception_source);
		}
```
Shouldn't the two lines be swapped ? o_O

Reply:
of course, which should also be the reason for the current SIGSEGVS in CI; please adjust this and - after verify that the SIGSEGVS are fixed, do another push adding the `#if 0` back again (also for the definition of last_exception_source in common.h) when last_exception_source is currently only strdup'd and free'd (and add it to the "to be inspected later" list)

Reply:
Okay, finally works. I'm not sure to understand the last part of your sentence though ("when last_exception_source is currently..."). Should I just put back `#if 0` everywhere there was originally + on the lines I added to make this work ?

Reply:
yes, starting with the definition of `last_exception_source` (as this will verify that it isn't used and needed somewhere)

Reply:
Done.